src/badge/index.tsx

```diff
@@ -7,23 +7,31 @@ import clsx from 'clsx';
 import { getBaseProps } from '../internal/base-component';
 import useBaseComponent from '../internal/hooks/use-base-component';
 import { applyDisplayName } from '../internal/utils/apply-display-name';
+import WithNativeAttributes from '../internal/utils/with-native-attributes';
 import { BadgeProps } from './interfaces';
 import { getBadgeStyles } from './style';
 
 import styles from './styles.css.js';
 
 export { BadgeProps };
 
-export default function Badge({ color = 'grey', children, style, ...rest }: BadgeProps) {
+export default function Badge({ color = 'grey', children, style, nativeAttributes, ...rest }: BadgeProps) {
   const { __internalRootRef } = useBaseComponent('Badge', { props: { color } });
   const baseProps = getBaseProps(rest);
 
   const className = clsx(baseProps.className, styles.badge, styles[`badge-color-${color}`]);
 
   return (
-    <span {...baseProps} {...{ className }} ref={__internalRootRef} style={getBadgeStyles(style)}>
+    <WithNativeAttributes
+      tag="span"
+      nativeAttributes={nativeAttributes}
+      {...baseProps}
+      {...{ className }}
```

Comment:
Why does it go in the middle of other props? Is it expected to override `tag` or `nativeAttributes`?

Now you are probably safe, because both `baseProps` and `className` are clean, but in general this "spread props in the middle" pattern is prone to errors and it is better to avoid it from the start